src/Common/src/System/Drawing/KnownColorTable.cs

```diff
@@ -438,15 +450,19 @@ private static int FromWin32Value(int value)
         }
 #endif
 
-#if FEATURE_SYSTEM_EVENTS
-        private static void OnUserPreferenceChanging(object sender, UserPreferenceChangingEventArgs e)
+        private static void OnUserPreferenceChanging(object sender, EventArgs args)
         {
-            if (e.Category == UserPreferenceCategory.Color && s_colorTable != null)
+            PropertyInfo categoryProperty = args.GetType()?.GetProperty("Category", BindingFlags.Instance | BindingFlags.Public);
+
+            if (categoryProperty == null)
```

Comment:
How frequently is this callback invoked?  Presumably very infrequently, in which case this is fine, but if it might be more common this reflection code could get expensive.

Also, do we expect this will always find "Category"?  Assuming yes, can we Debug.Assert that categoryProperty is non-null?

Reply:
> How frequently is this callback invoked?

The callback will be invoked infrequently. I checked the api port telemetry percentage for UserPreferenceChangingEventArgs and UserPreferenceChangingEventHandler which its pretty low

> Also, do we expect this will always find "Category"?

Yes it should always be present.

I added the ```Debug.Assert(categoryProperty != null);```
should I still keep the if check  ```if (categoryProperty == null) return;``` ?

Reply:
I'm not sure you can assume that.  This is invoked for every Windows message WM_SETTINGCHANGE or WM_SYSCOLORCHANGE.  These may happen semi-frequently in some cases.  It probably wouldn't hurt to cache a delegate to the property-getter.  Let me see if I can find an example for you.

Reply:
Here's a similar case.
https://github.com/dotnet/coreclr/blob/f2b6bc8bf19f74d20e00083320d219627ad3f695/src/System.Private.CoreLib/shared/System/Resources/ResourceReader.Core.cs#L74-L83

You should be able to create an unbound delegate to the property getter and cache it in a static.  In the case you can't find the field you could assert or just have a delegate that returns something like -1.

Reply:
@ericstj How frequent is the WM_SETTINGCHANGE/WM_SYSCOLORCHANGE, really?

The MakeGenericMethod is adding unavoidable extra JITting to WinForms startup path. It's not like we need more JITting in the WinForms startup path.

On the runtime team we've spent a lot of time in the past two months to get rid of JITting on startup for this release and it's unfortunate when I see *more* JITting happening in traces because of code like this. We should be going in this opposite direction.

This MakeGenericMethod trick should only be used when it's critical (I guess the ResourceReader qualifies).